editcomponents.cgi

```diff
@@ -29,28 +31,31 @@ my $vars     = {};
 # so all actions point to the same page.
 $vars->{'doc_section'} = 'components.html';
 
+print $cgi->header();
+
 #
-# Preliminary checks:
+# Often used variables
 #
+my $user              = Bugzilla->login(LOGIN_REQUIRED);
+my $product_name      = trim($cgi->param('product')   || '');
+my $comp_name         = trim($cgi->param('component') || '');
+my $action            = trim($cgi->param('action')    || '');
+my $showbugcounts     = (defined $cgi->param('showbugcounts'));
+my $token             = $cgi->param('token');
+my $editcomponents    = $user->in_group('editcomponents');
+my $edittriageowners  = $editcomponents || $user->in_group('edittriageowners');
```

Comment:
Setting `edittriageowners` to true if `editcomponents` is true makes it harder to reason about relevant permissions, and is a footgun (as shown by this causing an issue later in the code).

Keep `edittriageowners` separate from `editcomponents` and explicitly check both if required.